dark/dna.py

```diff
@@ -68,12 +68,31 @@ def compareDNAReads(read1, read2, matchAmbiguous=True, gapChars=('-'),
     read2AmbiguousOffsets = []
     empty = set()
 
+    def _identicalMatch(a, b):
+        return a == b and len(AMBIGUOUS[a]) == 1
+
+    def _ambiguousMatch(a, b, matchAmbiguous):
+        return (matchAmbiguous and 
+                not _identicalMatch(a, b) and
+                AMBIGUOUS.get(a, empty) & AMBIGUOUS.get(b, empty))
+        """
+        Checks if two characters match ambiguously if matchAmbiguous is True. 
+        A match is an ambiguous match if it is not an identical match, but the 
+        sets of ambiguous characters overlap.
+        """
+
     for offset, (a, b) in enumerate(zip_longest(read1.sequence.upper(),
                                                 read2.sequence.upper())):
         # Use 'is not None' in the following to allow an empty offsets set
         # to be passed.
         if offsets is not None and offset not in offsets:
             continue
+        if a is not None and a not in gapChars:    
```

Comment:
I think this can just be

```python
if len(AMBIGUOUS[a]) > 1:
    read1AmbiguousOffsets.append(offset)
```
I.e., there's no need for the outer test.

Reply:
I think there is - I tried without the outer test first, but then every character that is None or a gapChar gives an error message, because they can't be found in the $ AMBIGUOUS dictionary.

Reply:
Ah, sorry, you're right.  I'd do this:

```python
if len(AMBIGUOUS.get(a, '')) > 1:
    read1AmbiguousOffsets.append(offset)
```

It's a slight hack: you ask for the default value of the empty string to be returned if the key isn't in the dictionary, and that has a length of zero.  Or if you want to be more explicit you could do

```python
if a in AMBIGUOUS and len(AMBIGUOUS[a]) > 1:
    read1AmbiguousOffsets.append(offset)
```

but that looks in the dictionary twice.

Reply:
That's really cool, to be able to write the same thing so much shorter :) done.